### DISCUSSION TITLE
Any way to host this on a web server?

### DISCUSSION THREAD

**Discussion by whackyhook** [mayooear/ai-pdf-chatbot-langchain#286] [General]
Hi guys, excellent tool which works really effectively.

If one wished to host this on the web so it could be remotely accessed, what would be the best way to go about this?

Any ideas much appreciated.

**Comment by KleiAliaj** [comment]
I am also trying to  depoy this on vercel but An error occurred while fetching the data. Please try again.
any luck with this

**Comment by groovybits** [comment]
Setting up the Dockerfile and putting it into Google cloudrun is cheap and easy, will cost you nothing to run it unless it becomes super popular :)

https://github.com/groovybits/gaib/blob/main/Dockerfile
https://github.com/groovybits/gaib/blob/main/cloudbuild.yaml
https://github.com/groovybits/gaib/blob/main/CLOUDRUN.md

Mostly covers it, but specific to my fork, should not require all the variables and really is simple. Ask chatGPT-4 and you will have it running in cloudrun rather fast.

**Reply by olserra** [reply]
Hi, could you help me on this? I am having some problems here with the persissions on IAM

**Comment by BioStarr** [comment]
Have aws machine with some public accessible IP.
Use nginx to route port 80 requests to localhost:3000.
Have fun.

**Reply by olserra** [reply]
Hi, could you help me do it to production instead of localhost?

**Comment by olserra** [comment]
Any work around to deploy in Vercel? I would like to stick with Vercel
